jobs/create.js

```diff
@@ -1,38 +1,53 @@
-const uuid = require('uuid');
-const AWS = require('aws-sdk');
+"use strict";
+
+const uuid = require("uuid");
+const AWS = require("aws-sdk");
 
 const dynamoDb = new AWS.DynamoDB.DocumentClient();
 
-module.exports.create = (event, context, callback) => {
-    const timestamp = new Date().getTime();
-    const data = JSON.parse(event.body);
-    const dynamoData = {
-        TableName: 'jobs',
-        Item: {
-          jobId: uuid.v1(),
-          jobName: data.jobName,
-          createdAt: timestamp,
-          updatedAt: timestamp,
-        },
-    };
-    dynamoDb.put(dynamoData, (error) => {
-        // handle potential errors
-        if (error) {
-          console.error(error);
-          callback(null, {
-            statusCode: error.statusCode || 501,
-            headers: { 'Content-Type': 'text/plain' },
-            body: 'Couldn\'t create the todo item.',
-          });
-          return;
-        }
-    
-        // create a response
-        const response = {
-          statusCode: 200,
-          body: JSON.stringify(dynamoData),
-        };
-        callback(null, response);
-    });
+module.exports.create = async (event, context, callback) => {
+  const timestamp = new Date().getTime();
+  const data = JSON.parse(event.body);
+  const dynamoData = {
+    TableName: "jobs",
+    Item: {
+      jobId: uuid.v1(),
+      jobName: data.jobName,
+      createdAt: timestamp,
+      updatedAt: timestamp
+    }
+  };
+  try {
+    let result = await dynamoDb.put(dynamoData).promise();
+    const error = result.$response.error;
+
+    if (error) {
+      // TODO: This is repeated elsewhere maybe abstract this out into a helper
+      // function??
+      console.error(error);
+      const response = {
+        statusCode: error.statusCode,
+        headers: { "Content-Type": "text/plain" },
+        body: "Couldn't fetch the jobs."
+      };
+      callback(null, response);
+      return;
+    }
 
-}
\ No newline at end of file
+    const response = {
+      statusCode: 200,
+      body: JSON.stringify(dynamoData)
```

Comment:
This returns an object that looks like:
```js
{
    "TableName": "jobs",
    "Item": {
        "jobId": "f1153850-52ec-11e9-ad3b-950ad64365c0",
        "jobName": "Joel's first try",
        "createdAt": 1553951322196,
        "updatedAt": 1553951322196
    }
}
```
How about we just do `body: JSON.stringify(dynamoData.Item)` so that we get just the database item?

Reply:
@jdd1260 done